core/transientstore/store.go

```diff
@@ -153,7 +153,7 @@ func (s *Store) Persist(txid string, blockHeight uint64,
 
 // GetTxPvtRWSetByTxid returns an iterator due to the fact that the txid may have multiple private
 // write sets persisted from different endorsers.
-func (s *Store) GetTxPvtRWSetByTxid(txid string, filter ledger.PvtNsCollFilter) (RWSetScanner, error) {
+func (s *Store) GetTxPvtRWSetByTxid(txid string, filter ledger.PvtNsCollFilter) RWSetScanner {
```

Comment:
The existing signature was good but in the function the error is not being checked. In the leveldb iterator, you have to check for the error explicitly - [see usage here](https://github.com/hyperledger/fabric/blob/2fdbafbed9fc1213a28e5980b8c334ce04c27dcb/core/ledger/kvledger/txmgmt/statedb/stateleveldb/stateleveldb.go#L296)

Perhaps, we should handle this right in the beginning in the leveldbhelper package and return error from there, so not to miss this in the upper layers.

Reply:
Ah. So we have an existing bug in the transient store then. Will fix and update the PR.

Reply:
We have `missing error check` at the `stateleveldb` pkg too except the newly introduced `GetFullDBScanner()`.